src/ctapipe/irf/select.py

```diff
@@ -0,0 +1,298 @@
+    ).tag(config=True)
+
+    gammaness_classifier = Unicode(
+        default_value="RandomForestClassifier",
+        help="Prefix of the classifier `_prediction` column",
+    ).tag(config=True)
+
+    quality_criteria = List(
+        Tuple(Unicode(), Unicode()),
+        default_value=[
+            ("multiplicity 4", "np.count_nonzero(tels_with_trigger,axis=1) >= 4"),
+            ("valid classifier", "RandomForestClassifier_is_valid"),
+            ("valid geom reco", "HillasReconstructor_is_valid"),
+            ("valid energy reco", "RandomForestRegressor_is_valid"),
+        ],
+        help=QualityQuery.quality_criteria.help,
+    ).tag(config=True)
+
+    rename_columns = List(
+        help="List containing translation pairs new and old column names"
+        "used when processing input with names differing from the CTA prod5b format"
+        "Ex: [('alt_from_new_algorithm','reco_alt')]",
+        default_value=[],
+    ).tag(config=True)
+
+    def normalise_column_names(self, events: Table) -> QTable:
+        keep_columns = [
+            "obs_id",
+            "event_id",
+            "true_energy",
+            "true_az",
+            "true_alt",
+        ]
+        standard_renames = {
+            "reco_energy": f"{self.energy_reconstructor}_energy",
+            "reco_az": f"{self.geometry_reconstructor}_az",
+            "reco_alt": f"{self.geometry_reconstructor}_alt",
+            "gh_score": f"{self.gammaness_classifier}_prediction",
+        }
+        rename_from = []
+        rename_to = []
+        # We never enter the loop if rename_columns is empty
+        for old, new in self.rename_columns:
+            if new in standard_renames.keys():
+                self.log.warning(
+                    f"Column '{old}' will be used as '{new}' "
+                    f"instead of {standard_renames[new]}."
+                )
+                standard_renames.pop(new)
+
+            rename_from.append(old)
+            rename_to.append(new)
+
+        for new, old in standard_renames.items():
+            if old in events.colnames:
+                rename_from.append(old)
+                rename_to.append(new)
+
+        # check that all needed reco columns are defined
+        for c in ["reco_energy", "reco_az", "reco_alt", "gh_score"]:
+            if c not in rename_to:
+                raise ValueError(
+                    f"No column corresponding to {c} is defined in "
+                    f"EventPreProcessor.rename_columns and {standard_renames[c]} "
+                    "is not in the given data."
+                )
+
+        keep_columns.extend(rename_from)
+        events = QTable(events[keep_columns], copy=False)
+        events.rename_columns(rename_from, rename_to)
+        return events
+
+    def make_empty_table(self) -> QTable:
+        """
+        This function defines the columns later functions expect to be present
+        in the event table.
+        """
+        columns = [
```

Comment:
this has much duplication and has the danger of missing something / changing it only in one place.
it also results in the wrong dtypes for many columns (as float64 is the default and that's wrong e.g. for obs_id and event_id).


Better directly define the columns:
```
from astropy.table import Column

columns = [
    Column(name="obs_id", dtype=np.uint64, description="..."),
    Column(name="true_energy", unit=u.TeV, description="..."),
   ...
]
```

Reply:
`obs_id` and `event_id` are the only columns not using float64, or am I forgetting something?

Reply:
The _valid are bool

Reply:
Yeah, good call. Added explicit dimensionless units to the gh score and weight too.